validators/deposit_resource.go

```diff
@@ -1,22 +1,32 @@
 package validators
 
 import (
+	"strings"
+
+	"github.com/santhosh-tekuri/jsonschema"
 	"github.com/sul-dlss-labs/taco/db"
-	"github.com/sul-dlss-labs/taco/generated/models"
 )
 
 // DepositResourceValidator validates the deposit resource request
 type DepositResourceValidator struct {
 	repository db.Database
+	schema     *jsonschema.Schema
 }
 
 // NewDepositResourceValidator creates a new instance of DepositResourceValidator
-func NewDepositResourceValidator(repository db.Database) *DepositResourceValidator {
-	return &DepositResourceValidator{repository: repository}
+func NewDepositResourceValidator(repository db.Database, schemaPath string) *DepositResourceValidator {
```

Comment:
why is this connected to db.Database?

Reply:
We will need to query the database in order to validate that the source_id is unique.